### PR TITLE
fix amplicon clipping

### PR DESCRIPTION
**Pull request by alethomas**
inlcude hard clipping for amplicon primers and exclude the hard clipped sequence from exporting to new fastq files

### CODE REVIEW

File: workflow/rules/read_clipping.smk [hunk 1]
```diff
@@ -32,6 +42,12 @@ rule clip_primer:
         cd {params.dir}
         bamclipper.sh -b {params.bam} -p {params.dir_depth}{input.bed} -n {threads} >> {params.dir_depth}{log} 2>&1
         cd {params.dir_depth}
-        samtools sort  -@ {threads} -n {output.clippedbam} -o {output.sortclippedbam}  >> {log} 2>&1
-        samtools fastq -@ {threads} {output.sortclippedbam} -1 {output.fq1} -2 {output.fq2}  >> {log} 2>&1
+        fgbio --sam-validation-stringency=LENIENT ClipBam -i {output.clippedbam} -o {output.hardclippedbam} -H true -r {input.ref_fasta} >> {log} 2>&1
```
Comment (alethomas) [inline workflow/rules/read_clipping.smk:45]: pretty sure there is a cleaner way for doing this

File: workflow/rules/read_clipping.smk [hunk 2]
```diff
@@ -32,6 +42,12 @@ rule clip_primer:
...
+        samtools sort  -@ {threads} -n {output.hardclippedbam} -o {output.sortclippedbam}  >> {log} 2>&1
+
+        samtools fastq -@ {threads} {output.hardclippedbam} -1 {output.fq1_initial} -2 {output.fq2_initial}  >> {log} 2>&1
+        zcat {output.fq1_initial} | paste - - - - | sort -k1,1 -t \" \" | tr \"\t\" \"\n\" > {output.fq1_sorted}
+        zcat {output.fq2_initial} | paste - - - - | sort -k1,1 -t \" \" | tr \"\t\" \"\n\" > {output.fq2_sorted}
+        gzip -c {output.fq1_sorted} > {output.fq1}
+        gzip -c {output.fq2_sorted} > {output.fq2}
```
Comment (alethomas) [inline workflow/rules/read_clipping.smk:52]: pretty sure there would be a cleaner way for that?

Comment (johanneskoester) [reply workflow/rules/read_clipping.smk:52]: Why are these additional cmds needed at all? The clipbam just changed the clips, the bam should be the same other than that, right? And before, it worked fine to just use samtools sort and samtools fastq.

Comment (alethomas) [reply workflow/rules/read_clipping.smk:52]: Will try again if I messed something up with the samtools sort but it didn't work when I had it like that and I included the additional lines

**Review by thomasbtf — APPROVED** [review]

**Pull request merged**